pkg/prometheus/metrics.go

```diff
@@ -15,6 +17,8 @@ const (
 	ErrorResult   = "error"
 )
 
+type Provider string
```

Comment:
Do we need this type here? I think you replaced this and this is just a leftover.

Reply:
indeed, removing